File: Plugins/Aspose-Cells-Java-for-Jython/asposecells/WorkingWithFiles/ConvertingWorksheetToSVG.py
from asposecells import Settings
from com.aspose.cells import Workbook
from com.aspose.cells import ImageFormat
from com.aspose.cells import ImageOrPrintOptions
from com.aspose.cells import SheetRender
from com.aspose.cells import SaveFormat



class ConvertingWorksheetToSVG:

    def __init__(self):
        dataDir = Settings.dataDir + 'WorkingWithFiles/ConvertingWorksheetToSVG/'
        
        saveFormat = SaveFormat

        workbook = Workbook(dataDir + "Book1.xls")

        #Convert each worksheet into svg format in a single page.
        imgOptions = ImageOrPrintOptions()
        imgOptions.setSaveFormat(saveFormat.SVG)
        imgOptions.setOnePagePerSheet(True)

        #Convert each worksheet into svg format
        sheetCount = workbook.getWorksheets().getCount()

        #for(i=0; i<sheetCount; i++)
        for i in range(sheetCount):
        
            sheet = workbook.getWorksheets().get(i)

            sr = SheetRender(sheet, imgOptions)

            pageCount = sr.getPageCount()
            #for (k = 0 k < pageCount k++)
            for k in range(pageCount):
            
                #Output the worksheet into Svg image format
                sr.toImage(k, dataDir + sheet.getName() + ".out.svg")
            
        

        # Print message
        print "Excel to SVG conversion completed successfully."
 
if __name__ == '__main__':        
    ConvertingWorksheetToSVG()

File: Examples/src/main/java/com/aspose/cells/examples/loading_saving/ConvertingWorksheetToSVG.java
package com.aspose.cells.examples.loading_saving;

import com.aspose.cells.*;
import com.aspose.cells.examples.Utils;

public class ConvertingWorksheetToSVG {

	public static void main(String[] args) throws Exception {

		// The path to the documents directory.
		String dataDir = Utils.getSharedDataDir(ConvertingWorksheetToSVG.class) + "loading_saving/";

		String path = dataDir + "Book1.xlsx";

		// Create a workbook object from the template file
		Workbook workbook = new Workbook(path);

		// Convert each worksheet into svg format in a single page.
		ImageOrPrintOptions imgOptions = new ImageOrPrintOptions();
		imgOptions.setSaveFormat(SaveFormat.SVG);
		imgOptions.setOnePagePerSheet(true);

		// Convert each worksheet into svg format
		int sheetCount = workbook.getWorksheets().getCount();

		for (int i = 0; i < sheetCount; i++) {
			Worksheet sheet = workbook.getWorksheets().get(i);

			SheetRender sr = new SheetRender(sheet, imgOptions);

			for (int k = 0; k < sr.getPageCount(); k++) {
				// Output the worksheet into Svg image format
				sr.toImage(k, path + sheet.getName() + k + "_out.svg");
			}
		}

		// Print message
		System.out.println("Excel to SVG conversion completed successfully.");

	}
}
